### PR TITLE
Multi-OS Engine support + moving of generic Java code to commons

### PR DESCRIPTION
**Pull request by augustweinbren**
## Description

This PR does not modify any existing behavior. It consists of the following changes: 

- Support for Multi-OS Engine added to build.gradle
- `commons`(to consist of shared iOS/android code) and `ios` modules created.
- Boiler plate code for `ios` module automatically generated.
-  `Models` package + parts of `API` package which do not use Android libraries moved to `commons` module.

## Related Issues

[#90] (note: this PR should not close the issue.)

### CODE REVIEW

**Comment by augustweinbren** [comment]
Although the Android app builds successfully, I am currently seeing the following error message when running the app.

E/FirebaseCrashlytics: Settings request failed.
    java.io.FileNotFoundException: https://firebase-settings.crashlytics.com/spi/v2/platforms/android/gmp/1:460393230596:android:a4512316c4b07001132051/settings?instance=d2ce15d64f057f5a768d8a0187b20776f9e4e11b&build_version=581&display_version=0.7.0&source=1
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:255)
        at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getInputStream(DelegatingHttpsURLConnection.java:211)
        at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:30)
        at com.google.firebase.crashlytics.internal.network.HttpGetRequest.execute(HttpGetRequest.java:80)
        at com.google.firebase.crashlytics.internal.settings.network.DefaultSettingsSpiCall.invoke(DefaultSettingsSpiCall.java:113)
        at com.google.firebase.crashlytics.internal.settings.SettingsController$1.then(SettingsController.java:199)
        at com.google.firebase.crashlytics.internal.settings.SettingsController$1.then(SettingsController.java:192)
        at com.google.android.gms.tasks.zzo.run(com.google.android.gms:play-services-tasks@@17.2.0:2)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at com.google.firebase.crashlytics.internal.common.ExecutorUtils$1$1.onRun(ExecutorUtils.java:64)
        at com.google.firebase.crashlytics.internal.common.BackgroundPriorityRunnable.run(BackgroundPriorityRunnable.java:27)
        at java.lang.Thread.run(Thread.java:923)

**Comment by hpsaturn** [comment]
Thanks @augustweinbren I'm goint to test and review it..

File: settings.gradle [hunk 1]
```diff
@@ -1 +1,3 @@
 include ':app', ':libtoolscommon'
+include ':ios'
+include ':commons'
```
Comment (hpsaturn) [inline settings.gradle:3]: That is wrong in my environment, the project build fine but Android Studio can not see these packages, please change it to:

`include ':app', ':libtoolscommon', ':ios', ':commons'`

**Review by hpsaturn — CHANGES_REQUESTED** [review]
Thanks for your PR, for now it build ok and works ok in a first review, I have only one request. For other side I changed the pull request target branch to devel.

**Pull request merged**